Plutarch/Api/V1.hs

```diff
@@ -173,46 +162,55 @@ deriving via (DerivePConstantViaData Plutus.ScriptPurpose PScriptPurpose) instan
 deriving via (DerivePConstantViaNewtype Plutus.Datum PDatum PData) instance (PConstant Plutus.Datum)
 
 newtype PRedeemer (s :: S) = PRedeemer (Term s PData)
-  deriving (PlutusType, PIsData) via (DerivePNewtype PRedeemer PData)
+  deriving (PlutusType, PIsData, PEq) via (DerivePNewtype PRedeemer PData)
 
 instance PUnsafeLiftDecl PRedeemer where type PLifted PRedeemer = Plutus.Redeemer
 deriving via (DerivePConstantViaNewtype Plutus.Redeemer PRedeemer PData) instance (PConstant Plutus.Redeemer)
 
 newtype PDatumHash (s :: S) = PDatumHash (Term s PByteString)
-  deriving (PlutusType, PIsData) via (DerivePNewtype PDatumHash PByteString)
+  deriving (PlutusType, PIsData, PEq, POrd) via (DerivePNewtype PDatumHash PByteString)
 
 instance PUnsafeLiftDecl PDatumHash where type PLifted PDatumHash = Plutus.DatumHash
 deriving via (DerivePConstantViaNewtype Plutus.DatumHash PDatumHash PByteString) instance (PConstant Plutus.DatumHash)
 
 newtype PStakeValidatorHash (s :: S) = PStakeValidatorHash (Term s PByteString)
-  deriving (PlutusType, PIsData) via (DerivePNewtype PStakeValidatorHash PByteString)
+  deriving (PlutusType, PIsData, PEq, POrd) via (DerivePNewtype PStakeValidatorHash PByteString)
 
 instance PUnsafeLiftDecl PStakeValidatorHash where type PLifted PStakeValidatorHash = Plutus.StakeValidatorHash
-deriving via (DerivePConstantViaNewtype Plutus.StakeValidatorHash PStakeValidatorHash PByteString) instance (PConstant Plutus.StakeValidatorHash)
+deriving via
+  (DerivePConstantViaNewtype Plutus.StakeValidatorHash PStakeValidatorHash PByteString)
+  instance
+    (PConstant Plutus.StakeValidatorHash)
 
 newtype PRedeemerHash (s :: S) = PRedeemerHash (Term s PByteString)
-  deriving (PlutusType, PIsData) via (DerivePNewtype PRedeemerHash PByteString)
+  deriving (PlutusType, PIsData, PEq, POrd) via (DerivePNewtype PRedeemerHash PByteString)
 
 instance PUnsafeLiftDecl PRedeemerHash where type PLifted PRedeemerHash = Plutus.RedeemerHash
-deriving via (DerivePConstantViaNewtype Plutus.RedeemerHash PRedeemerHash PByteString) instance (PConstant Plutus.RedeemerHash)
+deriving via
+  (DerivePConstantViaNewtype Plutus.RedeemerHash PRedeemerHash PByteString)
+  instance
+    (PConstant Plutus.RedeemerHash)
 
 newtype PValidatorHash (s :: S) = PValidatorHash (Term s PByteString)
-  deriving (PlutusType, PIsData) via (DerivePNewtype PValidatorHash PByteString)
+  deriving (PlutusType, PIsData, PEq, POrd) via (DerivePNewtype PValidatorHash PByteString)
 
 instance PUnsafeLiftDecl PValidatorHash where type PLifted PValidatorHash = Plutus.ValidatorHash
-deriving via (DerivePConstantViaNewtype Plutus.ValidatorHash PValidatorHash PByteString) instance (PConstant Plutus.ValidatorHash)
+deriving via
+  (DerivePConstantViaNewtype Plutus.ValidatorHash PValidatorHash PByteString)
+  instance
+    (PConstant Plutus.ValidatorHash)
 
 ---------- Value
 
 newtype PTokenName (s :: S) = PTokenName (Term s PByteString)
   deriving newtype (Semigroup, Monoid)
```

Comment:
Why does this have a `Semigroup` and `Monoid` instance but the others don't btw?

Reply:
This doesn't even make sense. Token names have a limit of 32 bytes. I propose removing the instance because it's almost bogus.

Reply:
I agree with that. Removed.

I noticed that the plutus ledger api also doesn't implement semigroup/monoid for most of these (if any) types.